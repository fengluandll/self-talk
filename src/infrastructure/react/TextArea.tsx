import React, {
  FC,
  HTMLProps,
  InputHTMLAttributes,
  useEffect,
  useRef
} from "react";
import AutosizeTextArea from "react-autosize-textarea";
import styled from "styled-components";
import { boxShadow } from "./style";

const StyledTextArea: FC<any> = styled(AutosizeTextArea)`
  ${boxShadow};
  box-sizing: border-box;
  border: none;
  outline: none;
  width: 100%;
  color: inherit;
  font: inherit;
  resize: none;
  padding: 1em;
  border-radius: 0.5em;

  &::placeholder {
    color: grey;
  }
`;

export const TextArea = (
  props: InputHTMLAttributes<HTMLTextAreaElement> &
    HTMLProps<HTMLTextAreaElement>
) => {
  const ref = useRef<HTMLTextAreaElement | null>(null);

  useEffect(() => {
    if (ref.current) {
      ref.current.focus();
    }
  });

  return <StyledTextArea async={true} ref={ref} {...props} />;
};
