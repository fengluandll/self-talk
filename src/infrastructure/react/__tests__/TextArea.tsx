import React from "react";
import { create } from "react-test-renderer";
import { TextArea } from "../TextArea";

it("renders", () => {
  expect(
    create(<TextArea onSubmit={() => undefined} />).toJSON()
  ).toMatchSnapshot();
});
