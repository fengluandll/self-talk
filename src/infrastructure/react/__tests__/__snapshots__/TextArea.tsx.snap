// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<textarea
  async={true}
  className="sc-bdVaJa cPXXOd"
  onChange={[Function]}
  onKeyDown={[Function]}
  rows={1}
/>
`;
